*** Imakefile~	Tue Aug 17 13:59:02 1993
--- Imakefile	Tue Aug 17 14:03:59 1993
***************
*** 29,34 ****
--- 29,38 ----
  #endif
  #endif
  
+ #ifdef i386BsdArchitecture
+ 	DEFINES = -DBSD  -DI386BSD $(DEFGDB)
+ #endif
+ 
  #ifdef PegasusArchitecture
  DEFINES = -DBSD $(DEFGDB) -DNEED_STRSTR
  #endif
*** calldbx.c~	Tue Aug 17 14:04:13 1993
--- calldbx.c	Tue Aug 17 14:05:02 1993
***************
*** 77,82 ****
--- 77,86 ----
  #include	<termio.h>
  #endif
  
+ #ifdef I386BSD
+ #include      <sys/ioctl_compat.h>
+ #endif
+ 
  #include	"global.h"
  
  #ifdef SYSV 
