*** general/latex.bug.old	Fri Feb 21 17:22:08 1992
--- general/latex.bug	Fri Feb 21 18:16:32 1992
***************
*** 1,6 ****
  LATEST VERSIONS OF FILES -- LaTeX Version 2.09
  
! ADDENDUM.TEX   8 Apr 90    ARTICLE.DOC/STY   14 Jan 92   LATEX.BUG  14 Jan 92
  IDX.TEX       22 Jan 85    ART10.DOC/STY     14 Jan 92   LATEX.INS   1 Dec 91
  LABLST.TEX     5 Jan 85    ART11.DOC/STY     14 Jan 92   LOCAL.GID  24 Sep 85
  LATEX.TEX     14 Jan 92    ART12.DOC/STY     14 Jan 92
--- 1,6 ----
  LATEST VERSIONS OF FILES -- LaTeX Version 2.09
  
! ADDENDUM.TEX   8 Apr 90    ARTICLE.DOC/STY   14 Jan 92   LATEX.BUG  21 Feb 92
  IDX.TEX       22 Jan 85    ART10.DOC/STY     14 Jan 92   LATEX.INS   1 Dec 91
  LABLST.TEX     5 Jan 85    ART11.DOC/STY     14 Jan 92   LOCAL.GID  24 Sep 85
  LATEX.TEX     14 Jan 92    ART12.DOC/STY     14 Jan 92
***************
*** 11,17 ****
  MAKEINDEX.TEX 17 Feb 87    BK12.DOC/STY      14 Jan 92
  SAMPLE.TEX     9 Apr 86    FLEQN.DOC/STY     04 Nov 91
  SFONTS.TEX    25 Nov 91    IFTHEN.DOC/STY    16 Jun 91
! SLITEX.TEX    14 Jan 92    LETTER.DOC/STY    04 Nov 91
  SMALL.TEX      5 Jul 85    LEQNO.DOC/STY     29 Sep 91
  SPLAIN.TEX    14 Jan 92    MAKEIDX.DOC/STY   21 Oct 91
  TESTPAGE.TEX  20 Jul 87    OPENBIB.DOC/STY   14 Jan 92
--- 11,17 ----
  MAKEINDEX.TEX 17 Feb 87    BK12.DOC/STY      14 Jan 92
  SAMPLE.TEX     9 Apr 86    FLEQN.DOC/STY     04 Nov 91
  SFONTS.TEX    25 Nov 91    IFTHEN.DOC/STY    16 Jun 91
! SLITEX.TEX    14 Jan 92    LETTER.DOC/STY    21 Feb 92
  SMALL.TEX      5 Jul 85    LEQNO.DOC/STY     29 Sep 91
  SPLAIN.TEX    14 Jan 92    MAKEIDX.DOC/STY   21 Oct 91
  TESTPAGE.TEX  20 Jul 87    OPENBIB.DOC/STY   14 Jan 92
***************
*** 778,784 ****
  
  176. \maketitle or other commands that internally invoke a list
  environment would not be detected before \begin{document}. (Corrected
! 26 Aug 91.)
  
  177. Moved the \SLiTeX logo to latex.tex so that it can be used in
  LaTeX as well. (Change made 29 Sep 91.)
--- 778,786 ----
  
  176. \maketitle or other commands that internally invoke a list
  environment would not be detected before \begin{document}. (Corrected
! 26 Aug 91.) NOTE: THIS CHANGE AFFECTS ALL STYLES AND STYLE OPTIONS
! THAT CHANGE THE \begin{document} COMMAND! IT IS NECESSARY TO ADD
! \@noskipsecfalse.
  
  177. Moved the \SLiTeX logo to latex.tex so that it can be used in
  LaTeX as well. (Change made 29 Sep 91.)
***************
*** 1172,1179 ****
  
  76. Added explicit setting of \topskip in article, book, and report
  document styles. This is important for sizes 11pt and 12pt to avoid
! ``overfull vbox while output is active'' messages. (Suggested by
  Friedrich Haubensak. Change made 14 Jan 92.)
  
  
                           CHANGES TO THE MANUAL
--- 1174,1187 ----
  
  76. Added explicit setting of \topskip in article, book, and report
  document styles. This is important for sizes 11pt and 12pt to avoid
! ``overfull \vbox while \output is active'' messages. (Suggested by
  Friedrich Haubensak. Change made 14 Jan 92.)
+ 
+ 77. Change no. 176 above had also to be done in the letter document
+ style. (Corrected 21 Feb 92.)
+ 
+ 78. Fix no. 59 caused leading blanks in the argument of \begin{letter}
+ not to be ignored. (Reported by Will Partain. Corrected 21 Feb 92.)
  
  
                           CHANGES TO THE MANUAL
*** doc/letter.doc.old	Fri Feb 21 17:22:31 1992
--- doc/letter.doc	Fri Feb 21 17:29:38 1992
***************
*** 1,8 ****
! % LETTER STANDARD DOCUMENT STYLE -- Released 05 November 91
  %    for LaTeX version 2.09
  % Copyright (C) 1991 by Leslie Lamport
  
! \typeout{Standard Document Style `letter' <05 Nov 91>.}
  
  % PREPARING A FOREIGN LANGUAGE VERSION:
  %
--- 1,8 ----
! % LETTER STANDARD DOCUMENT STYLE -- Released 21 February 92
  %    for LaTeX version 2.09
  % Copyright (C) 1991 by Leslie Lamport
  
! \typeout{Standard Document Style `letter' <21 Feb 92>.}
  
  % PREPARING A FOREIGN LANGUAGE VERSION:
  %
***************
*** 485,495 ****
  
  
  %% FMi & RmS: added \leavevmode to catch empty argument, 16 Aug 91
  
  \long\def\letter#1{\newpage
    \c@page\@ne
    \interlinepenalty=200 % smaller than the TeXbook value
!   \@processto{\leavevmode #1}}
  
  \def\endletter{\stopletter\@@par\pagebreak\@@par
    \if@filesw
--- 485,496 ----
  
  
  %% FMi & RmS: added \leavevmode to catch empty argument, 16 Aug 91
+ %% RmS ...and added \ignorespaces, 21 Feb 92
  
  \long\def\letter#1{\newpage
    \c@page\@ne
    \interlinepenalty=200 % smaller than the TeXbook value
!   \@processto{\leavevmode\ignorespaces #1}}
  
  \def\endletter{\stopletter\@@par\pagebreak\@@par
    \if@filesw
***************
*** 548,553 ****
--- 549,557 ----
  % This will also work with old lfonts if no other style defines
  % |\process@table|.
  %
+ % 92/02/21 RmS: added \@noskipsecfalse setting in conformance with
+ %               change in latex.tex.
+ %
  \def\document{\endgroup
    \@colht\textheight  \@colroom\textheight \vsize\textheight
     \columnwidth\textwidth \@clubpenalty\clubpenalty
***************
*** 564,570 ****
    \def\do##1{\let ##1\@notprerr}%
    \@preamblecmds
    \let\do\noexpand
!   \@normalsize\everypar{}}
  
  % Redefine \enddocument so it outputs the last page of labels,
  % if necessary.
--- 568,574 ----
    \def\do##1{\let ##1\@notprerr}%
    \@preamblecmds
    \let\do\noexpand
!   \@normalsize\everypar{}\@noskipsecfalse}
  
  % Redefine \enddocument so it outputs the last page of labels,
  % if necessary.
*** sty/letter.sty.old	Fri Feb 21 17:22:40 1992
--- sty/letter.sty	Fri Feb 21 17:36:05 1992
***************
*** 1,8 ****
! % LETTER STANDARD DOCUMENT STYLE -- Released 05 November 91
  %    for LaTeX version 2.09
  % Copyright (C) 1991 by Leslie Lamport
  
! \typeout{Standard Document Style `letter' <05 Nov 91>.}
  
  \def\ccname{cc}
  \def\enclname{encl}
--- 1,8 ----
! % LETTER STANDARD DOCUMENT STYLE -- Released 21 February 92
  %    for LaTeX version 2.09
  % Copyright (C) 1991 by Leslie Lamport
  
! \typeout{Standard Document Style `letter' <21 Feb 92>.}
  
  \def\ccname{cc}
  \def\enclname{encl}
***************
*** 252,258 ****
  \long\def\letter#1{\newpage
    \c@page\@ne
    \interlinepenalty=200
!   \@processto{\leavevmode #1}}
  
  \def\endletter{\stopletter\@@par\pagebreak\@@par
    \if@filesw
--- 252,258 ----
  \long\def\letter#1{\newpage
    \c@page\@ne
    \interlinepenalty=200
!   \@processto{\leavevmode\ignorespaces #1}}
  
  \def\endletter{\stopletter\@@par\pagebreak\@@par
    \if@filesw
***************
*** 314,320 ****
    \def\do##1{\let ##1\@notprerr}%
    \@preamblecmds
    \let\do\noexpand
!   \@normalsize\everypar{}}
  
  \def\enddocument{\@checkend{document}\newpage\begingroup
  \if@filesw \immediate\closeout\@mainaux
--- 314,320 ----
    \def\do##1{\let ##1\@notprerr}%
    \@preamblecmds
    \let\do\noexpand
!   \@normalsize\everypar{}\@noskipsecfalse}
  
  \def\enddocument{\@checkend{document}\newpage\begingroup
  \if@filesw \immediate\closeout\@mainaux
----------------------------------------------------------------

