*** crypt.c.orig	Mon Jul 12 06:29:34 1993
--- crypt.c	Sun Jul 11 07:38:45 1993
***************
*** 66,74 ****
--- 66,80 ----
  	(void) dup (fildes);		  /* standard in from file */
  	(void) dup (pipefd[1]);		  /* connect to pipe */
  	(void) fprintf (stderr, " ");
+ #ifdef BDES
+ 	(void) execl (CRYPT_PATH, "bdes", "-d", "-k", KeyWord, 0);
+ 	(void) fprintf (stderr, "execl(%s, \"bdes\", \"-dk\",  %s, 0) in creadfile() failed",
+ 			CRYPT_PATH, KeyWord);
+ #else
  	(void) execl (CRYPT_PATH, "crypt", KeyWord, 0);
  	(void) fprintf (stderr, "execl(%s, \"crypt\", %s, 0) in creadfile() failed",
  			CRYPT_PATH, KeyWord);
+ #endif
  	exit (-127);
      }
      else				  /* else parent */
***************
*** 159,167 ****
--- 165,179 ----
  	(void) dup (pipefd[0]);			  /* connect to pipe input */
  	(void) dup (fildes);			  /* standard out to file */
  	(void) fprintf (stderr, " ");
+ #ifdef BDES
+ 	(void) execl (CRYPT_PATH, "bdes", "-k",  KeyWord, 0);
+ 	(void) fprintf (stderr, "execl(%s, \"bdes\", \"-k\", %s, 0) in cwritefile() failed",
+ 			CRYPT_PATH, KeyWord);
+ #else
  	(void) execl (CRYPT_PATH, "crypt", KeyWord, 0);
  	(void) fprintf (stderr, "execl(%s, \"crypt\", %s, 0) in cwritefile() failed",
  			CRYPT_PATH, KeyWord);
+ #endif
  	exit (-127);
      }
      else				  /* else parent */
