*** Makefile.orig	Tue Jan 25 02:13:33 1994
--- Makefile	Mon Feb 14 21:29:15 1994
***************
*** 14,19 ****
  	talk tcopy tee telnet tftp time tip tn3270 touch tput tr true tset \
  	tsort tty ul uname unexpand unifdef uniq unvis users uudecode uuencode \
  	vacation vgrind vi vis vmstat w wall wc what whereis which who whoami \
! 	whois window write xargs xinstall xstr yacc yes
  	
  .include <bsd.subdir.mk>
--- 14,19 ----
  	talk tcopy tee telnet tftp time tip tn3270 touch tput tr true tset \
  	tsort tty ul uname unexpand unifdef uniq unvis users uudecode uuencode \
  	vacation vgrind vi vis vmstat w wall wc what whereis which who whoami \
! 	whois window write xargs xinstall xstr yacc yes pcvt
  	
  .include <bsd.subdir.mk>
