? 
--- c-auto.h	Tue Apr 12 02:26:17 1994
+++ /usr/local.old/src/web2c-6.1/web2c/lib/c-auto.h	Tue Apr 12 01:33:26 1994
@@ -152,7 +152,7 @@
 /* #undef const */
 
 /* Define as the proper declaration for yytext.  */
-#define DECLARE_YYTEXT "extern char  *yytext;"
+#define DECLARE_YYTEXT extern char  *yytext;
 
 /* Define if you have dirent.h.  */
 #define DIRENT 1
--- ourpaths.c	Tue Apr 12 01:19:59 1994
+++ /usr/local.old/src/web2c-6.1/web2c/lib/ourpaths.c	Sun Dec  5 12:48:26 1993
@@ -2,7 +2,7 @@
 
 #include "config.h"
 
-#include <kpathsea/tex-file.h>
+#include <kpathsea/filefmt.h>
 #include <kpathsea/fontmap.h>
 #include <kpathsea/pathsearch.h>
 
@@ -78,7 +78,7 @@
   /* Look for it.  Don't use the kpse_find_glyph stuff, since we don't
      have the dpi available separately, and anyway we don't care about
      having pktogf run MakeTeXPK, etc.  */
-  found = kpse_path_search (path, name, 0);
+  found = kpse_path_search (path, name);
 
   /* If we didn't find it, and we're looking for a font, maybe it's
      an alias defined in a mapping file.  This duplicates most of
@@ -88,10 +88,10 @@
                  || path_index == PKFILEPATH))
     {
       string *mapped_names;
-      static hash_table_type fontmap;
+      static map_type fontmap = NULL;
       
       /* Fault in the mapping if necessary.  */
-      if (fontmap.size == 0)
+      if (!fontmap)
         fontmap = map_create (path);
       
       /* Now look for our filename in the mapping.  */
@@ -101,7 +101,7 @@
           string mapped_name;
           while ((mapped_name = *mapped_names++) && !found)
             {
-              found = kpse_path_search (path, mapped_name, 0);
+              found = kpse_path_search (path, mapped_name);
             }
         }
     }
